portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,13 +20,39 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Date;
+
+import java.util.ArrayList;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    
+    String jsonStr = convertToJsonUsingGson();
```

Comment:
Option 1:

```java
    SomeType someName = someFunction();
    doSomethingWith(someName);
```

Option 2:

```java
    doSomethingWith(someFunction());
```

I'd say that you want to use option 1:

  * When someFunction() is very complicated.
  * When the return type is weird
  * When `someName` helps making the code more understandable, for example by explaining the intent of something where someFunction() is not obvious, e.g. `float objectDistance = sqrt(dot(object1.position().sub(object2.position())));`.

But when these don't apply, sometimes option 2 can be more clear.